### PR TITLE
chore(mvn): update dependency versions in poms

### PR DESCRIPTION
**Pull request by maximilianhuber**
In this commit not all dependencies were forced to their latest state. The changes are listed in the following table

| dependency | old version | new version | latest version |
| --- | --- | --- | --- |
| ch.qos.logback:logback-classic | 1.1.2 | 1.2.3 | |
| ch.qos.logback:logback-core | 1.1.2 | 1.2.3 | |
| com.google.code.gson:gson | 2.6.2 | 2.8.0 | |
| com.google.guava:guava | 18.0 | 21.0 | 22.0-android |
| com.jcraft:jsch | 0.1.51 | 0.1.54 | |
| com.liferay.portal:portal-service | 6.2.1 | 6.2.2 | 6.2.5 |
| com.liferay.portal:util-bridges | 6.2.1 | 6.2.2 | 6.2.5 |
| com.liferay.portal:util-java | 6.2.1 | 6.2.2 | 6.2.5 |
| com.liferay.portal:util-taglib | 6.2.1 | 6.2.2 | 6.2.5 |
| commons-cli:commons-cli | 1.2 | 1.4 | |
| com.tngtech.java:junit-dataprovider | 1.10.0 | 1.12.0 | |
| com.tngtech.jgiven:jgiven-junit | 0.9.4 | 0.15.1 | |
| javax.mail:mail | 1.4 | 1.4.7 | 1.5.0-b01 |
| javax.portlet:portlet-api | 2.0 | **2.0** | 3.0.0 |
| javax.servlet.jsp:jsp-api | 2.0 | **2.0** | 2.2.1-b03 |
| javax.servlet:servlet-api | 2.4 | **2.4** | 3.0-alpha-1 |
| javax.servlet:servlet-api | 2.5 | **2.4** | 3.0-alpha-1 |
| junit:junit | 4.11 | 4.12 | |
| org.apache.commons:commons-csv | 1.0 | 1.4 | |
| org.apache.httpcomponents:httpcore | 4.3.3 | 4.4.6 | |
| org.apache.poi:poi | 3.9 | 3.16 | |
| org.apache.poi:poi-ooxml | 3.9 | 3.16 | |
| org.apache.thrift:libthrift | 0.9.3 | 0.9.3 | 0.10.0 |
| org.apache.velocity:velocity | 1.6.2 | 1.7 | |
| org.jetbrains:annotations | 13.0 | 15.0 | |
| org.mockito:mockito-all | 1.10.5 | 1.10.19 | 2.0.2-beta |
| org.slf4j:slf4j-api | 1.7.1 | 1.7.25 | 1.8.0-alpha2 |
| org.slf4j:slf4j-api | 1.7.7 | 1.7.25 | 1.8.0-alpha2 |
| org.slf4j:slf4j-log4j12 | 1.7.1 | 1.7.25 | 1.8.0-alpha2 |
| org.slf4j:slf4j-log4j12 | 1.7.7 | 1.7.25 | 1.8.0-alpha2 |
| org.springframework:spring-beans | 4.1.5.RELEASE | 4.3.8.RELEASE | |
| org.springframework:spring-context | 4.1.5.RELEASE | 4.3.8.RELEASE | |
| org.springframework:spring-webmvc | 4.1.5.RELEASE | 4.3.8.RELEASE | |

## 

to list the dependencies which need an update use
```
mvn versions:display-dependency-updates | grep '\->'  | sed 's/\[INFO\] *//' | sort | uniq
```
to force the update for all dependencies use
```
mvn versions:use-latest-releases
```

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -89,7 +89,7 @@
         <dependency>
             <groupId>javax.mail</groupId>
             <artifactId>mail</artifactId>
-            <version>1.4</version>
+            <version>1.5.0-b01</version>
```
Comment (alexbrdn) [inline backend/pom.xml]: If `-b01` designates a beta version, I'd suggest going to the closest stable version

File: frontend/sw360-portlet/pom.xml [hunk 2]
```diff
@@ -128,19 +128,19 @@
...
             <artifactId>servlet-api</artifactId>
-            <version>2.4</version>
+            <version>2.5</version>
             <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>javax.servlet.jsp</groupId>
             <artifactId>jsp-api</artifactId>
-            <version>2.0</version>
+            <version>2.2.1-b03</version>
```
Comment (alexbrdn) [inline frontend/sw360-portlet/pom.xml]: If `-b03` designates a beta version, I'd suggest going to the closest stable version

**Comment by mcjaeger** [comment]
possible, we could also go for 6.2.5 for the liferay dependencies.

**Comment by maximilianhuber** [comment]
The inherited dependencies now are matching the one provided by liferay and thus should also be consistent.

@alexbrdn can you please retest this PR?

@mcjaeger compiling with a liferay version >6.2.2 leads always to a build error like:
![2017-06-06_16 06 16](https://cloud.githubusercontent.com/assets/1187050/26833217/25510aba-4ad2-11e7-8bfa-83a3efcac5e0.png)
```
[...]
[INFO] svc-schedule ....................................... SUCCESS [  1.128 s]                                                                                                                                   
[INFO] backend-utils ...................................... SUCCESS [ 17.109 s]                                                                                                                                   
[INFO] frontend ........................................... SUCCESS [  0.029 s]                                                                                                                                   
[INFO] SW360 Layout Template .............................. SUCCESS [  0.624 s]                                                                                                                                   
[INFO] sw360-portlet ...................................... FAILURE [ 21.773 s]                                                                                                                                   
[INFO] sw360-theme Theme .................................. SKIPPED                                                                                                                                               
[INFO] ------------------------------------------------------------------------                                                                                                                                   
[INFO] BUILD FAILURE                                                                                                                                                                                              
[INFO] ------------------------------------------------------------------------                                                                                                                                   
[INFO] Total time: 05:11 min                                                                                                                                                                                      
[INFO] Finished at: 2017-06-06T16:06:08+02:00                                                                                                                                                                     
[INFO] Final Memory: 129M/1737M                                                                                                                                                                                   
[INFO] ------------------------------------------------------------------------                                                                                                                                   
[ERROR] Failed to execute goal com.liferay.maven.plugins:liferay-maven-plugin:6.2.5:build-css (default) on project sw360-portlet: Some problems were encountered while processing the POMs:                       
[ERROR] [ERROR] Failed to determine Java version for profile jdk8 @ com.google.guava:guava:[unknown-version], /home/mhuber/.m2/repository/com/google/guava/guava/21.0/guava-21.0.pom, line 155, column 14         
[ERROR] [ERROR] Failed to determine Java version for profile jdk8 @ com.google.guava:guava-parent:21.0, /home/mhuber/.m2/repository/com/google/guava/guava-parent/21.0/guava-parent-21.0.pom, line 286, column 14 
[ERROR] : 2 problems were encountered while building the effective model for com.google.guava:guava:21.0                                                                                                          
[ERROR] [ERROR] Failed to determine Java version for profile jdk8 @ com.google.guava:guava:[unknown-version], /home/mhuber/.m2/repository/com/google/guava/guava/21.0/guava-21.0.pom, line 155, column 14         
[ERROR] [ERROR] Failed to determine Java version for profile jdk8 @ com.google.guava:guava-parent:21.0, /home/mhuber/.m2/repository/com/google/guava/guava-parent/21.0/guava-parent-21.0.pom, line 286, column 14 
[ERROR] -> [Help 1]                                                                                                                                                                                               
[ERROR]                                                                                                                                                                                                           
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.                                                                                                                               
[ERROR] Re-run Maven using the -X switch to enable full debug logging.                                                                                                                                            
[ERROR]                                                                                                                                                                                                           
[ERROR] For more information about the errors and possible solutions, please read the following articles:                                                                                                         
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException                                                                                                                          
[ERROR]                                                                                                                                                                                                           
[ERROR] After correcting the problems, you can resume the build with the command                                                                                                                                  
[ERROR]   mvn <goals> -rf :sw360-portlet  
```

**Comment by adrian-evo** [comment]
Basic functionality works fine, except #450

**Comment by mcjaeger** [comment]
I tried this branch now, did the following steps:

1. install via fresh vagrant
2. do the 4 config steps (user assoc, passwd policies, private pages, public pages)
3. Import the test users form the vagrant project in shared
4. login with admin@sw360.org -> vulnerabilities tab works
5. login with user@sw360.org -> vulnerabilities tab works

could it be that there is an issue with the wrong lar files?

**Pull request merged**